source/Date.cpp

```diff
@@ -49,23 +52,57 @@ Date::Date(int day, int month, int year)
 // Convert a date to a string.
 const string &Date::ToString() const
 {
+	string dateFormat = Preferences::DateFormat();
+
+	if(dateFormat != dateFormatInUse)
+	{
+		dateFormatInUse = dateFormat;
+		str = "";
+	}
+
 	// Because this is a somewhat "costly" operation, cache the result. The
 	// cached value is discarded if the date is changed.
-	if(date && str.empty())
+	if((date && str.empty()) || dateFormat != dateFormatInUse)
```

Comment:
```suggestion
	if((date && str.empty()) || dateFormat != dateFormatInUse)
```
The last one will never happen because of the code in front of it...

Reply:
This confuses me. There's no change to the code.

Reply:
```
        if(dateFormat != dateFormatInUse)
	{
		dateFormatInUse = dateFormat;
		str = "";
	}
```
Because if this.

Reply:
Now there is a change

Reply:
Do this!!!

Reply:
Resolved by 111b0229621bed50e192df52ee68ef441b988b35